File: src/main/java/onion/compiler/toolbox/Paths.java
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler.toolbox;

import java.io.File;

/**
 * @author Kota Mizushima
 * Date: 2005/06/17
 */
public class Paths {
  private Paths(){}
  public static String nameOf(String path){
    return new File(path).getName();
  }
  public static String cutExtension(String path){
    String name = nameOf(path);
    return name.substring(0, name.lastIndexOf('.'));
  }
}

File: src/main/scala/onion/compiler/toolbox/Paths.scala
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2016-, Kota Mizushima, All rights reserved.  *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler.toolbox

import java.io.File

/**
 * @author Kota Mizushima
 *        
 */
object Paths {
  def nameOf(path: String): String = new File(path).getName

  def cutExtension(path: String): String = {
    val name = nameOf(path)
    name.lastIndexOf('.') match {
      case n if n < 0 => name
      case n => name.substring(0, n)
    }
  }
}
